src/Tools/dotnet-monitor/ConfigurationJsonWriter.cs

```diff
@@ -217,25 +220,64 @@ private void ProcessSection(IConfigurationSection section, bool includeChildSect
                     {
                         if (child.GetChildren().Any())
                         {
-                            ProcessChildren(child, includeChildSections, redact);
+                            ProcessChildren(child, includeChildSections, redact,  showSources: showSources);
                         }
                         else
                         {
                             WriteValue(child.Value, redact);
+
+                            string comment = GetConfigurationProvider(child, showSources);
+
+                            if (comment.Length > 0)
+                            {
+                                _writer.WriteCommentValue(comment);
+                            }
                         }
                     }
 
                     _writer.WriteEndArray();
                 }
                 else
                 {
-                    ProcessChildren(section, includeChildSections, redact);
+                    ProcessChildren(section, includeChildSections, redact, showSources: showSources);
                 }
             }
             else if (!canWriteChildren)
             {
                 WriteValue(section.Value, redact);
+
+                string comment = GetConfigurationProvider(section, showSources);
+
+                if (comment.Length > 0)
+                {
+                    _writer.WriteCommentValue(comment);
+                }
+            }
+        }
+
+        private string GetConfigurationProvider(IConfigurationSection section, bool showSources)
+        {
+            if (showSources)
+            {
+                var configurationProviders = ((IConfigurationRoot)_configuration).Providers.Reverse();
+
+                StringBuilder comment = new StringBuilder();
+
+                foreach (var provider in configurationProviders)
+                {
+                    provider.TryGet(section.Path, out string value);
+
+                    if (!string.IsNullOrEmpty(value))
+                    {
+                        comment.Append(provider.ToString());
```

Comment:
Since we are not actually merging multiple providers here, can we just use a string instead of StringBuilder?